tests/test_browser.py

```diff
@@ -1822,6 +1822,10 @@ def test_emscripten_main_loop_and_blocker(self):
     for args in [[], ['-s', 'USE_PTHREADS', '-s', 'PROXY_TO_PTHREAD']]:
       self.btest('emscripten_main_loop_and_blocker.cpp', '0', args=args)
 
+  @requires_threads
+  def test_emscripten_main_loop_and_blocker_exit(self):
+    self.btest_exit('emscripten_main_loop_and_blocker.cpp', 0)
```

Comment:
this test already exists, doesn't this add a duplicated run of it?

Reply:
This on has the "_exit" suffix because it tests with `btest_exit` rather than `btest` which itself used `EXIT_RUNTIME=1`.  

The reason this is interesting/important to this change is that it shows that you the main loop can run and then stop and the runtime will exit once the loop as stopped running.  With the old `exitRuntime` variable once the main loop was started the runtime could *never* again exit (at least not using the existing APIs). 

With the push/pop system the runtime knows when the last loops is done and it can naturally exit at that point,